beta-src/src/utils/map/drawArrow.ts

```diff
@@ -22,190 +22,208 @@ export default function drawArrow(
-      const fromTerritoryEl: SVGSVGElement = d3
-        .select(`#${fromTerritoryName}-territory`)
-        .node();
-      const toTerritoryEl: SVGSVGElement = d3
-        .select(`#${toTerritoryName}-territory`)
-        .node();
+  const fromTerritoryName = Territory[unitTerritory];
+  const fromTerritoryEl: SVGSVGElement = d3
+    .select(`#${fromTerritoryName}-territory`)
+    .node();
+  let unit = d3.select(`#${fromTerritoryName}-unit`).node();
+  let unitX;
+  let unitY;
+  let unitW;
+  let unitH;
 
-      if (fromTerritoryEl && toTerritoryEl && receiverUnit && unitSlotEl) {
-        unitSlotEl = unitSlotEl.parentNode;
-        receiverUnit = receiverUnit.parentNode;
-        const unitSlotElBBox = unitSlotEl.getBBox();
-        const receiverUnitBBox = receiverUnit.getBBox();
-        const unitSlotElX =
-          Number(unitSlotEl.getAttribute("x")) + unitSlotElBBox.x;
-        const receiverSlotElX =
-          Number(receiverUnit.getAttribute("x")) + receiverUnitBBox.x;
-        const receiverSlotElY =
-          Number(receiverUnit.getAttribute("y")) + receiverUnitBBox.y;
+  if (unit && fromTerritoryEl) {
+    unit = unit.parentNode;
+    const fromTerritoryX = Number(fromTerritoryEl.getAttribute("x"));
+    const fromTerritoryY = Number(fromTerritoryEl.getAttribute("y"));
+    ({ x: unitX, y: unitY, width: unitW, height: unitH } = unit.getBBox());
+    unitX = fromTerritoryX + Number(unit.getAttribute("x")) + unitX;
+    unitY = fromTerritoryY + Number(unit.getAttribute("y")) + unitY;
+    switch (receiverType) {
+      case "arrow": {
+        const arrow: SVGLineElement = d3
+          .selectAll(`.arrow__${receiverIdentifier}`)
+          .node();
 
-        x1 = Number(fromTerritoryEl.getAttribute("x")) + unitSlotElX;
-        x2 = Number(toTerritoryEl.getAttribute("x")) + receiverSlotElX;
+        if (arrow) {
+          x1 = unitX;
+          y1 = unitY;
+          const attachPoint = 0.6;
+          const arrowX1 = Number(arrow.getAttribute("x1"));
+          const arrowY1 = Number(arrow.getAttribute("y1"));
+          const arrowX2 = Number(arrow.getAttribute("x2"));
+          const arrowY2 = Number(arrow.getAttribute("y2"));
+          const run = arrowX2 - arrowX1;
+          const rise = arrowY2 - arrowY1;
+          x2 = arrowX1 + run * attachPoint;
+          y2 = arrowY1 + rise * attachPoint;
+        }
+        break;
+      }
+      case "unit": {
+        const toTerritoryName = Territory[receiverIdentifier];
+        let receiverUnit = d3.select(`#${toTerritoryName}-unit`).node();
+        const toTerritoryEl: SVGSVGElement = d3
+          .select(`#${toTerritoryName}-territory`)
+          .node();
+
+        if (toTerritoryEl && receiverUnit) {
+          receiverUnit = receiverUnit.parentNode;
+
+          const {
+            x: rUnitX,
+            y: rUnitY,
+            width: rUnitW,
+            height: rUnitH,
+          } = receiverUnit.getBBox();
```

Comment:
Nit: Can we inline this like we did above.

Reply:
We can't inline this as prettier automatically formats it this way.